### ISSUE TITLE
na_ontap_disk_options error autoassign

### ISSUE THREAD

**Issue by lerd2805** [ansible-collections/netapp.ontap#50]
### Summary

When I try to change the autoassign option I get an error. How na_ontap_disk_options works?

### Component Name

na_ontap_disk_options

### Ansible Version

```console
ansible 2.9.18
```

### ONTAP Collection Version

```console
netapp.ontap:21.16.0
```

### ONTAP Version

```console
NetApp Release 9.10.1
```

### Playbook

```console
---
- hosts: localhost
  gather_facts: false
  collections:
    - netapp.ontap
  tasks:
  - name: Disable Disk Auto Assign to {{ netapp_cluster_name }}-01
    na_ontap_disk_options:
      https: true
      validate_certs: false
      hostname: "{{ netapp_cluster_mgmt_ip }}"
      username: "{{ netapp_username }}"
      password: "{{ netapp_password }}"
      node: "{{ netapp_cluster_name }}-01"
      autoassign: false
      
  - name: Disable Disk Auto Assign to {{ netapp_cluster_name }}-02
    na_ontap_disk_options:
      https: true
      validate_certs: false
      hostname: "{{ netapp_cluster_mgmt_ip }}"
      username: "{{ netapp_username }}"
      password: "{{ netapp_password }}"
      node: "{{ netapp_cluster_name }}-02"
      autoassign: false
```

### Steps to Reproduce

Run the playbook on a netapp cluster 9.10.1

### Expected Results

Be able to modify the autoassign option

### Actual Results

```console
{
  "module_stdout": "",
  "module_stderr": "/var/lib/awx/venv/ansible/lib/python3.6/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning,\n/var/lib/awx/venv/ansible/lib/python3.6/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning,\nTraceback (most recent call last):\n  File \"/tmp/ansible_na_ontap_disk_options_payload_rj4vief1/ansible_na_ontap_disk_options_payload.zip/ansible_collections/netapp/ontap/plugins/module_utils/netapp_module.py\", line 273, in get_modified_attributes\n  File \"/tmp/ansible_na_ontap_disk_options_payload_rj4vief1/ansible_na_ontap_disk_options_payload.zip/ansible_collections/netapp/ontap/plugins/module_utils/netapp_module.py\", line 59, in cmp\nTypeError: '>' not supported between instances of 'str' and 'bool'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1646241507.730173-24894-128560760775834/AnsiballZ_na_ontap_disk_options.py\", line 100, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1646241507.730173-24894-128560760775834/AnsiballZ_na_ontap_disk_options.py\", line 92, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1646241507.730173-24894-128560760775834/AnsiballZ_na_ontap_disk_options.py\", line 41, in invoke_module\n    run_name='__main__', alter_sys=True)\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_na_ontap_disk_options_payload_rj4vief1/ansible_na_ontap_disk_options_payload.zip/ansible_collections/netapp/ontap/plugins/modules/na_ontap_disk_options.py\", line 172, in <module>\n  File \"/tmp/ansible_na_ontap_disk_options_payload_rj4vief1/ansible_na_ontap_disk_options_payload.zip/ansible_collections/netapp/ontap/plugins/modules/na_ontap_disk_options.py\", line 168, in main\n  File \"/tmp/ansible_na_ontap_disk_options_payload_rj4vief1/ansible_na_ontap_disk_options_payload.zip/ansible_collections/netapp/ontap/plugins/modules/na_ontap_disk_options.py\", line 155, in apply\n  File \"/tmp/ansible_na_ontap_disk_options_payload_rj4vief1/ansible_na_ontap_disk_options_payload.zip/ansible_collections/netapp/ontap/plugins/module_utils/netapp_module.py\", line 275, in get_modified_attributes\nTypeError: TypeError(\"'>' not supported between instances of 'str' and 'bool'\",), key: autoassign, value: 'on', desired: False\n",
  "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
  "rc": 1,
  "_ansible_no_log": false,
  "changed": false
}
```

**Comment by carchi8py** [comment]
We have issue DEVOPS-4737 for this issue.

**Comment by lonico** [comment]
Interestingly, ONTAP 9.10.1 returns
```
"bkg_firmware_update": "on", "autocopy": "on", "autoassign": "on",
```
while 9.8 returns
```
"bkg_firmware_update": true, "autocopy": true, "autoassign": false
```

**Comment by lonico** [comment]
ONTAP confirmed the change in 9.10.1 is intentional.  We're updating the module to support both types.

**Comment by lonico** [comment]
fixed in 21.18.0
